frontend/snapps/httpsnapp/src/index.ts

```diff
@@ -0,0 +1,74 @@
+import fetch from 'node-fetch';
+
+type PublicKey = string;
+
+type Signature = { field: string; scalar: string };
+
+type Signed<Signable> = {
+  publicKey: PublicKey;
+  signature: Signature;
+  payload: Signable;
+};
+
+type Method = 'GET' | 'POST';
+
+type QueryParams = {};
+
+type Response = {
+  ok?: Signed<string>;
+  error?: Error;
+};
+
+/**
+ * Currently, WIP. Sends either a GET or POST request to the HTTPSnapps Oracle API which will
+ * returned a signed response from an official oracle key.
+ *
+ * @param _method
+ * @param _url
+ * @param _params
+ * @param _postData
+ * @returns
+ */
+export const verify = async (
+  method: Method,
+  url: string,
+  params: QueryParams,
+  _postData: string
+): Promise<Response> => {
+  // Convert QueryParams to query string
+  const qs = Object.keys(params)
+    .map((key) => `${key}=${params[key]}`)
+    .join('&');
+
+  // Temporary URL, Oracle API runs on localhost for now
```

Comment:
seems like we should adjust the API to take the URI prefix. I don't know what's the typical way to do this these days in JS-land.

We could do

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps({ url: "https://localhost:3000" }).verify
```

or

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps.makeVerify({ url: "https://localhost:3000" })
```

or something else?

Reply:
Something like this is probably most common, if you don't expect the params to change:

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps.verify("https://localhost:3000")
```

or the object param works too:

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps.verify({ url: "https://localhost:3000" })
```

(It's easy to switch between the two without a breaking change if needed in either case, by checking the param type.)

Reply:
Added a way to pass in the prefix using some currying. The naming/usage feels a little off so please let me know if there are other improvements to be added. The usage can be seen in the `simple.test.ts` file.